.github/workflows/lint.yml

```diff
@@ -0,0 +1,61 @@
+name: Lint
+
+on: [push]
```

Comment:
Do we want to break this up like earlier, so that say if I push Markdown files, only Markdown linting is done? For the rest, I'm okay with everything being executed in one go. Otherwise, I'd much rather there is a distinction between Markdown vs non-Markdown linting (which is usually code or code adjacent). The reason is that otherwise if I'm editing docs, the CI pipeline runs for a very long time and blocks merging.

The other reason is on our Github plan we have a limited number of free execution minutes and if we're running very long Actions sequences on every push, it's very wasteful in terms of minutes executed. So being a bit more intelligent on what Actions are executed on what push will help us optimise that.